### PR TITLE
[Refactor] ShowForumThread uses Twig

### LINKED ISSUE #3779: [Refactor] Twig engine for showForumThreads in Discussion Forum
**What problem are you trying to solve with Submitty**
Currently, the Discussion Forum renders HTML directly from the View file. This has to be shifted to be using the Twig template engine for consistency with the rest of the Submitty project.

This issue focuses on the `showForumThreads` method.

**Describe the way you'd like to solve this problem**
Refactor the app/views/forum/ForumThreadView.php to use the Twig template engine and put the respective templates in the app/templates/forum directory.

### PR DESCRIPTION
**Pull request by xprilion**
### Please check if the PR fulfills these requirements:
* [x] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

### What is the current behavior?
Closes #3779  

### What is the new behavior?
The following Discussion Forum view now uses Twig- 

`showForumThreads` 

### Other information?
Is this a breaking change? - None discovered yet.
How did you test - Proper working checked manually and by inspecting difference in HTML rendered before and after Twig template.

### CODE REVIEW

File: site/app/views/forum/ForumThreadView.php [hunk 1]
```diff
@@ -108,281 +108,232 @@ public function searchResult($threads){
...
+            "currentCourse" => $currentCourse,
+            "generate_post_content" => $generatePostContent,
+            "display_option" => $display_option
+        ]);
         return $return;
-	}
+    }
 
-    public function generatePostList($currentThread, $posts, $unviewed_posts, $currentCourse, $includeReply = false, $threadExists = false, $display_option = 'time', $categories = [], $cookieSelectedCategories = [], $cookieSelectedThreadStatus = [], $cookieSelectedUnread = [], $currentCategoriesIds = [], $isCurrentFavorite = false) {
+    public function generatePostList($currentThread, $posts, $unviewed_posts, $currentCourse, $includeReply = false, $threadExists = false, $display_option = 'time', $categories = [], $cookieSelectedCategories = [], $cookieSelectedThreadStatus = [], $cookieSelectedUnread = [], $currentCategoriesIds = [], $isCurrentFavorite = false, $render=true) {
```
[the remark this answers is not in the document]
Comment (xprilion) [reply site/app/views/forum/ForumThreadView.php]: @sak6lab probably should open a new issue for unused variables in ForumThreadView.php as other functions too have it, and some of them require changes at places not within the Forum, hence including those changes in Forum refactor may leave future contributors confused!

[the remark this answers is not in the document]
Comment (xprilion) [reply site/app/views/forum/ForumThreadView.php]: This function, for example, is used in the Electronic Grader...so changing that file while working on Forum refactor is not very intuitive and the exact requirement of the change may not be obvious from the issue/pr title

**Comment by bmcutler** [comment]
@xprilion  please resolve merge conflicts

**Comment by xprilion** [comment]
@bmcutler done!

**Pull request merged**